lib/AdminPages/Pages/NewsletterEditor.php

```diff
@@ -21,22 +22,41 @@ class NewsletterEditor {
   /** @var UserFlagsController */
   private $user_flags;
 
+  /** @var WooCommerceHelper */
+  private $woocommerce_helper;
+
   /** @var WPFunctions */
   private $wp;
 
   function __construct(
     PageRenderer $page_renderer,
     SettingsController $settings,
     UserFlagsController $user_flags,
+    WooCommerceHelper $woocommerce_helper,
     WPFunctions $wp
   ) {
     $this->page_renderer = $page_renderer;
     $this->settings = $settings;
     $this->user_flags = $user_flags;
+    $this->woocommerce_helper = $woocommerce_helper;
     $this->wp = $wp;
   }
 
   function render() {
+    $newsletter_id = (isset($_GET['id']) ? (int)$_GET['id'] : 0);
+    $woocommerce_template_id = (int)$this->settings->get('woocommerce.transactional_email_id', null);
+    if (
+      $woocommerce_template_id
+      && $newsletter_id === $woocommerce_template_id
+      && (
+        !$this->woocommerce_helper->isWooCommerceActive()
```

Comment:
Seeing this line I think when WC is not active then `page=mailpoet-settings#woocommerce` won't even exist. I guess the spec doesn't say what to do in such case, does it?

Reply:
Added redirect in JS when the user tries to access the WooCommerce tab without WooCommerce being active. I specifically didn't use `this.tabs('basics)` because that would only change the tab without changing the URL.